pyiron_atomistics/atomistics/structure/factories/materialsproject.py

```diff
@@ -152,4 +162,18 @@ def by_id(material_id: Union[str, int], api_key=None) -> Atoms:
         if api_key is not None:
             rest_kwargs["api_key"] = api_key
         with MPRester(**rest_kwargs) as mpr:
-            return pymatgen_to_pyiron(mpr.get_structure_by_material_id(material_id))
+            if final:
+                return pymatgen_to_pyiron(
+                    mpr.get_structure_by_material_id(
+                        material_id, final, conventional_unit_cell
+                    )
+                )
+            else:
+                return [
+                    pymatgen_to_pyiron(mpr_structure)
+                    for mpr_structure in (
+                        mpr.get_structure_by_material_id(
+                            material_id, final, conventional_unit_cell
```

Comment:
```suggestion
                            material_id=material_id, final=final, conventional_unit_cell=conventional_unit_cell
```

Whenever we use an external tool, it's always good to set the arguments explicitly, because they might change the order or add/remove certain arguments and we might not notice it.